edx/edx-enterprise
# ENT-196 Add check that SAP SF configuration works for connecting…

… to SAP SF

I still need to do some test/quality cleanup but wanted to get this on your radar.

*Introduced new fields on the SAPSuccessFactorsEnterpriseCustomerConfiguration that will
be needed for authenticating
*Added field in the admin that calls a client that performs fetching an access token from SAP,
this field is a simple checkmark icon to indicate whether the fetch succeeds or fails

**Description:** Describe in a couple of sentence what this PR adds

**JIRA:** Link to JIRA ticket

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**

Create an SAP SuccessFactors Global Configuration with the following values:
Oauth api path: learning/oauth-api/rest/v1/token
Other fields can be anything

Create an SAP SuccessFactorsEnterpriseCustomerConfiguration with the following values:
Sapsf base url: http://private-894b2-successfactors.apiary-mock.com/
Client ID: t1
Client Secret: 52d85f24492ff918dcd1935a45a9d34acd79e249fe331753c6e0105b109e034a
SAP Company ID: t1
SAP User ID: Usr1
The sap access token field should display a green checkmark icon.

Create another SAP SuccessFactorsEnterpriseCustomerConfiguration with random values. The sap access token field display a red dash icon.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
